app/controllers/wishlists_controller.rb

```diff
@@ -22,11 +23,27 @@ def create
   end
 
   def destroy
-    @wishlist = Wishlist.find(params[:id])
     authorize! :destroy, @wishlist
     @wishlist.destroy!
     redirect_to wishlists_path, notice: t(".success")
-  rescue CanCan::AccessDenied, ActiveRecord::RecordNotFound, ActiveRecord::RecordNotDestroyed
+  rescue CanCan::AccessDenied, ActiveRecord::RecordNotDestroyed
+    redirect_to wishlists_path, alert: t("generic_error")
+  end
+
+  def edit
+    authorize! :edit, @wishlist
+  rescue CanCan::AccessDenied
+    redirect_to wishlists_path, alert: t("generic_error")
+  end
+
+  def update
+    authorize! :update, @wishlist
+    if @wishlist.update_attributes(wishlist_params)
```

Comment:
That's just a note. You could use `update!` to avoid using `if`, and then handle the bad path in a `rescue` block catching `ActiveRecord::RecordInvalid`. But just my personal habit, I tend to avoid using `if` as much as possible.